pkg/operator/ceph/object/notification/controller.go

```diff
@@ -0,0 +1,251 @@
+	})
+}
+
+func addNotificationReconciler(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+	logger.Info("successfully started")
+
+	var CBNKind = reflect.TypeOf(cephv1.CephBucketNotification{}).Name()
+
+	var controllerTypeMeta = metav1.TypeMeta{
+		Kind:       CBNKind + "Labels",
+		APIVersion: fmt.Sprintf("%s/%s", cephv1.CustomResourceGroup, cephv1.Version),
+	}
+
+	// Watch for changes on the OBC CRD object
+	err = c.Watch(&source.Kind{Type: &cephv1.CephBucketNotification{TypeMeta: controllerTypeMeta}}, &handler.EnqueueRequestForObject{}, opcontroller.WatchControllerPredicate())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Reconcile reads that state of the cluster for a CephBucketNotification object and makes changes based on the state read
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileNotifications) Reconcile(context context.Context, request reconcile.Request) (reconcile.Result, error) {
+	// workaround because the rook logging mechanism is not compatible with the controller-runtime logging interface
+	reconcileResponse, err := r.reconcile(request)
+	if err != nil {
+		logger.Errorf("failed to reconcile %v", err)
+	}
+
+	return reconcileResponse, err
+}
+
+func (r *ReconcileNotifications) reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// fetch the CephBucketNotification instance
+	cephBN := &cephv1.CephBucketNotification{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, cephBN)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debugf("CephBucketNotification %q resource not found. Ignoring since resource must be deleted.", request.NamespacedName)
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, errors.Wrapf(err, "failed to retrieve CephBucketNotification %q", request.NamespacedName)
+	}
+
+	// find the namespace for the ceph cluster (may be different than the namespace of the topic CR)
+	// Make sure a CephCluster is present otherwise do nothing
+	cephCluster, isReadyToReconcile, cephClusterExists, reconcileResponse := opcontroller.IsReadyToReconcileNoName(
+		r.client,
+		r.context,
+		controllerName,
+	)
+	if !isReadyToReconcile {
+		// This handles the case where the Ceph Cluster is gone and we want to delete that CR
+		if !cephBN.GetDeletionTimestamp().IsZero() && !cephClusterExists {
+			// Return and do not requeue. Successful deletion.
+			return reconcile.Result{}, nil
+		}
+		logger.Debug("Ceph cluster not yet present.")
+		return reconcileResponse, nil
+	}
+	r.clusterSpec = &cephCluster.Spec
+
+	// DELETE: the CR was deleted
+	if !cephBN.GetDeletionTimestamp().IsZero() {
```

Comment:
If the resource has no finalizer you could just decide not to reconcile on DELETE event and let kube doing the GC Or is a placeholder for later?

Reply:
> Is the `CephBucketNotification` in a different namespace than the ceph cluster?

the notifications and topics belong should be in the namespace of the application (the bucket)

> If the resource has no finalizer you could just decide not to reconcile on DELETE event and let kube doing the GC Or is a placeholder for later?

its a placeholder. will do that part in later PR